### ISSUE TITLE
not able to serve ionic

### ISSUE THREAD

**Issue by feadrick** [Resgrid/Dispatch#1]
im not able to run ionic command ionic serve
im using node v16.2.0

![not able to compile](https://user-images.githubusercontent.com/26769295/134040581-fb919160-4e8e-4443-aa29-e38f19473d21.png)

**Comment by ucswift** [comment]
Hey @feadrick, yes I haven't pushed the latest code to this repo yet. We have an internal git server used for active dev, then I sync that over to github. Let me see if I can get that pushed up for you.

**Comment by feadrick** [comment]
thats great thanks

**Comment by ucswift** [comment]
Just wanted to give you a heads up.

The last 2 weeks we implemented a new v4 api and implemented a new shared library for all of our Angular & Ionic apps. The V4 API is live in our hosted environment (https://api.resgrid.com/index.html?urls.primaryName=Resgrid%20API%20V4) but not yet pushed out to the Core GitHub repo. The new shared library ngx-resgridlib is v4 api only and is still in early'ish dev which we are doing with the Dispatch app, plan is once those are stable it'll be pushed to Github/NPM but currently there is too much thrashing of the library.

TL;DR of the above, current state of Dispatch repo has a lib req that is currently only accessible via our internal npm repo. But I'm working on pushing the last v3 API version of the Dispatch code base for you, should be done latter tonight.

**Comment by ucswift** [comment]
@feadrick code updated https://github.com/Resgrid/Dispatch/commit/605ee486b534a1c6f041e4da21e0301721890143 note this version does not have Ionic in it, just Angular. The npm run commands (i.e. npm run build, npm run start) should be good. Tested them before the push. This is the code version we had deployed out to https://dispatch.resgrid.com/ up until this last weekend. It isn't 100% feature complete, but has almost all the mainline features/functionality done.

**Comment by feadrick** [comment]
hi Shawn,is there any command that i should know to run the app
because  when i pull the project ,i cannot run ionic serve anymore ,
i got this error -->[ERROR] Sorry! ionic serve can only be run in an Ionic project directory.

**Comment by ucswift** [comment]
Yea `npm run build` and `npm run start`, it's a standard Angular project now, the readme is not updated yet.

**Comment by feadrick** [comment]
oo i see,thanks

**Comment by feadrick** [comment]
hi shawn,when i run npm run build i got this error and i notice that no plugin  folder inside the project custom directory
![WhatsApp Image 2021-09-21 at 12 17 58](https://user-images.githubusercontent.com/26769295/134111007-3ace546b-820f-4a24-89bb-0dd5033231a4.jpeg)

**Comment by ucswift** [comment]
Oh dang, I missed those files, pushed https://github.com/Resgrid/Dispatch/commit/aa339a802161dd504672b1fa117e22cb37901c1a can you pull and try again.

**Comment by feadrick** [comment]
hi shawn, which version of the resgrid core should i use for this dispatch project,
right now im able to start the dispatch but after i login  and inspect the element, i got 404 from api that i setup
1.  api/v3/Voice/GetDepartmentVoiceSettings 
2.api/v3/Dispatch/GetNewCallForm

when i check the documentation ,
there is no path for voice and GetNewCallFrom

**Comment by ucswift** [comment]
Yea I didn't think of that, those are new API's and not in the Core codebase yet. That functionality will need to be disabled in the Dispatch app to get it working for the current code base. Voip should be easy, just comment out `this.store.dispatch(new VoiceActions.GetVoipInfo());` in the Dashboard page. For the new call form you'll need to disable the `getNewCallForm` in the home provider.

Everything will be in sync for our December release, right now the Dispatch app is still in dev. so everything will be cut and releases done at that time.

**Comment by feadrick** [comment]
the newcallForm is refer to the getnewcall form right?so what should id assign to the new call form is it an empty String "" ?

**Comment by ucswift** [comment]
you can assign it to a new formdataresult to be safe so ` NewCallForm: new FormDataResult()`

![image](https://user-images.githubusercontent.com/410646/134185370-c6a3ce95-ee61-466d-9d12-9fc3cbd44337.png)

**Comment by feadrick** [comment]
hi shawn,im able to run the project now thanks for your help,
however im facing different problem where request from dispatch blocked by Cors policy api
![not load data](https://user-images.githubusercontent.com/26769295/134274445-707ffee4-403f-4913-8f47-fee03169b108.png)

it happen on this api endpoint Calls/GetActiveCalls
i try to fix by on api by doing some code modification on api startup.cs
app.UseCors(x => x
		.AllowAnyOrigin()
		.AllowAnyMethod()
		.AllowAnyHeader()
		.WithExposedHeaders("*"));

the request is still blocked by cors policy.

**Comment by ucswift** [comment]
So the API itself should allow everything:
https://github.com/Resgrid/Core/blob/master/Web/Resgrid.Web.ServicesCore/Startup.cs#L369

You might want to check the error logs in the Application logs in the Windows event viewer, sometimes there could be an underlying error and it 'presents' as an CORS issue. There also could be something being set by the hosting service or web server that is setting the CORS value itself. 

If can hook a Visual Studio debugger up to the running api process when you try and call that you might get more detail there as well, if there is some underlying issue.

**Comment by ucswift** [comment]
Dispatch has been migrated to the v4 api and it's install is docker container based now. If you have any further trouble please open a new issue.
